Rakefile

```diff
@@ -71,6 +77,91 @@ def re_run_basic_smoke_test
+    jenkins_result = uri.open(redirect: false)
+  rescue OpenURI::HTTPError => e
+    abort "ERROR: Could not get lastSuccessfulBuild for #{CURRENT_BRANCH} of puppet-agent: '#{e.message}'"
+  end
+
+  begin
+    jenkins_result_parsed = JSON.parse(jenkins_result.read)
+  rescue JSON::ParserError => e
+    abort "ERROR: Could not get lastSuccessfulBuild's valid json for #{CURRENT_BRANCH}: '#{e.message}'"
+  end
+
+  begin
+    jenkins_result_parameters = jenkins_result_parsed['actions'][0]['parameters']
+  rescue
+    abort "ERROR: Could not get lastSuccessfulBuild's actions or parameters for #{CURRENT_BRANCH}"
+  end
+
+  jenkins_result_parsed['actions'][0]['parameters']
+end
+
+def lookup_passing_puppetagent_sha(my_jenkins_passing_json)
+  begin
+    my_jenkins_passing_json.find{|x| x['name'] == 'SUITE_COMMIT'}['value']
+  rescue
+    abort "ERROR: Could not get lastSuccessfulBuild's SUITE_COMMIT value for #{CURRENT_BRANCH}"
+  end
+end
+def lookup_passing_puppet_sha(my_jenkins_passing_json)
+  begin
+    my_jenkins_passing_json.find{|x| x['name'] == 'puppet_COMPONENT_COMMIT'}['value']
+  rescue
+    abort "ERROR: Could not get lastSuccessfulBuild's puppet_COMPONENT_COMMIT value for #{CURRENT_BRANCH}"
+  end
+end
+
+def git_passing_puppet_version
+   #FIXME: this should be updated when the package yaml file contains the metadata we need
+   #  we have to replace the hyphens with dots because, vanagon
+  `cd #{PUPPET_SUBMODULE_PATH}; git describe`.strip.gsub(/-/,'.')
+end
+
+def replace_puppet_pins(passing_puppetagent_sha, passing_puppet_version)
+  # read beaker options hash from its file
+  puts("replacing puppet sha and version in #{BEAKER_OPTIONS_FILE} " \
+       "with agent sha: #{passing_puppetagent_sha} and puppet version: #{passing_puppet_version}")
+  beaker_options_from_file = eval(File.read(BEAKER_OPTIONS_FILE))
+  # add puppet version values
+  beaker_options_from_file[:puppet_version]       = passing_puppet_version
+  beaker_options_from_file[:puppet_build_version] = passing_puppetagent_sha
+  File.write(BEAKER_OPTIONS_FILE, beaker_options_from_file.pretty_inspect)
+end
+
+namespace :puppet_submodule do
+  desc 'update puppet submodule commit'
+  task :update_puppet_version do
+    #  ensure we fetch here, or the describe done later could be wrong
+    my_jenkins_passing_json = jenkins_passing_json_parsed
+    git_checkout_command = "cd #{PUPPET_SUBMODULE_PATH} && git fetch origin && " \
+      "git checkout #{lookup_passing_puppet_sha(my_jenkins_passing_json)}"
+    puts("checking out known passing puppet version in submodule: `#{git_checkout_command}`")
+    system(git_checkout_command)
+    # replace puppet version and sha pins in beaker options file
+    replace_puppet_pins(lookup_passing_puppetagent_sha(my_jenkins_passing_json), git_passing_puppet_version)
+  end
+  desc 'commit and push; CAUTION: WILL commit and push, upstream, local changes to the puppet submodule and acceptance options'
+  task :commit_push do
```

Comment:
do you need separate rake tasks for that?  or are you cool with just two `system` calls and you can test by commenting them out?